Mathlib/CategoryTheory/Sites/ObjectsCoverTop.lean

```diff
@@ -0,0 +1,163 @@
+/-
+Copyright (c) 2023 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.CategoryTheory.Sites.Sheaf
+
+/-! Objects which cover the terminal object
+
+In this file, given a site `(C, J)`, we introduce the notion of a family
+of objects `Y : I → C` which "cover the final object": this means
+that for all `X : C`, the sieve `Sieve.ofObjects Y X` is covering for `J`.
+When there is a terminal object `X : C`, then `J.ObjectsCoverTop Y`
+holds iff `Sieve.ofObjects Y X` is covering for `J`.
+
+We introduce a notion of compatible family of elements on objects `Y`
+and obtain `Presheaf.FamilyOfElementsOnObjects.IsCompatible.exists_unique_section`
+which asserts that if a presheaf of types is a sheaf, then any compatible
+family of elements on objects `Y` which cover the final object extends as
+a section of this presheaf.
+
+-/
+
+universe w v' v u' u
+
+namespace CategoryTheory
+
+open Limits
+
+variable {C : Type u} [Category.{v} C] (J : GrothendieckTopology C)
+  {A : Type u'} [Category.{v'} A]
+
+namespace GrothendieckTopology
+
+/-- A family of objects `Y : I → C` "covers the final object"
+  if for all `X : C`, the sieve `ofObjects Y X` is a covering sieve. -/
+def ObjectsCoverTop {I : Type*} (Y : I → C) : Prop :=
```

Comment:
What do you think of the shorter name
```suggestion
def CoversTop {I : Type*} (Y : I → C) : Prop :=
```

Reply:
Thanks for the suggestion. I have made the change.